File: BlueTeam/api/schemas/alert.py
from datetime import datetime
from pydantic import BaseModel


class AlertResponse(BaseModel):
    alert_id: str
    event_id: str
    timestamp: datetime
    severity: str
    risk_score: int
    attack_types: list[str]
    source_ips: list[str]
    rule_names: list[str]
    detection_count: int
    reasons: list[str]
    status: str
    created_at: datetime


class AlertListResponse(BaseModel):
    """
    Response model for GET /alerts.
    """
    persisted: bool = True
    status: str = "OK"
    message: str = "Alerts are successfully persisted in the database."
    alerts: list[AlertResponse] | None = None

File: BlueTeam/dashboard/src/types/alert.ts
export interface Alert {
  alert_id: string;
  timestamp: string;
  severity: 'SAFE' | 'SUSPICIOUS' | 'HIGH RISK' | string;
  risk_score: number;
  attack_types: string[];
  source_ips: string[];
  rule_names: string[];
  detection_count: number;
  reasons: string[];
  status?: string;
}

export interface AlertListResponse {
  persisted: boolean;
  status: string;
  message: string;
  alerts: Alert[] | null;
}

export interface WebSocketAlertMessage {
  type: 'security_alert' | string;
  alert_id: string;
  risk_level: string;
  risk_score: number;
  attack_types: string[];
  source_ips: string[];
  rules_triggered: string[];
  reasons: string[];
  timestamp: string;
}
